pkg/cluster/ensureendpoints.go

```diff
@@ -14,12 +19,35 @@ import (
 // subnets for storage account access, but only if egress lockdown is
 // not enabled.
 func (m *manager) ensureServiceEndpoints(ctx context.Context) error {
+	// Only add service endpoints to the subnet if egress lockdown is not enabled.
+	if m.doc.OpenShiftCluster.Properties.FeatureProfile.GatewayEnabled {
+		return nil
+	}
+
 	subnetIds, err := m.getSubnetIds()
 	if err != nil {
 		return err
 	}
 
-	return m.subnet.CreateOrUpdateFromIds(ctx, subnetIds, m.doc.OpenShiftCluster.Properties.FeatureProfile.GatewayEnabled)
+	for _, subnetId := range subnetIds {
+		r, err := arm.ParseResourceID(subnetId)
+		if err != nil {
+			return err
+		}
+		subnet, err := m.armSubnets.Get(ctx, r.ResourceGroupName, r.Parent.Name, r.Name, nil)
+		if err != nil {
+			return err
+		}
+		shouldUpdate := addEndpointsToSubnet(api.SubnetsEndpoints, &subnet.Subnet)
+		if !shouldUpdate {
+			continue
+		}
+		err = m.armSubnets.CreateOrUpdateAndWait(ctx, r.ResourceGroupName, r.Parent.Name, r.Name, subnet.Subnet, nil)
+		if err != nil {
+			return err
+		}
```

Comment:
Just a comment off the top of my head.

There is nothing wrong with this, but a slightly different approach that could also work would be to divide that in the following way: 
- first get all the subnets that should be updated (in a function returning a slice)
- pass those subnets that should be updated (return value from the previous function) to another function that would simply do _CreateOrUpdateAndWait_ for each value in the slice.

I don't think is much better than what the PR does, but it sticks to the principle of not iterating + doing stuff with what you are iterating, but instead, first do some processing and once that is processed update the values in Azure. Not asking to change that, just a comment about a pattern that sometimes can be useful, even to simplify unit tests in some cases (as you don't need to mock _m.armSubnets_ because you just want to test the function that does the business logic which is filtering the subnets).

🙂